Objects/object.c

```diff
@@ -2616,6 +2617,35 @@ PyUnstable_Object_EnableDeferredRefcount(PyObject *op)
 #endif
 }
 
+int
+PyUnstable_Object_IsUniqueTemporary(PyObject *op)
+{
+    if (!_PyObject_IsUniquelyReferenced(op)) {
```

Comment:
I think this just demonstrates that `_PyObject_IsUniquelyReferenced` is badly named, otherwise we wouldn't need a new function.

We need to check that the following are true:
* The object has an explicit (not borrowed) reference count of 1
* It is mortal
* It does not support deferred reclamation.

I think `_PyObject_IsUniquelyReferenced` does that, but it isn't clear from the name.

Reply:
Let's discuss the name for `_PyObject_IsUniquelyReferenced` in https://github.com/python/cpython/pull/133144